### PR TITLE
Add a new endpoint listCatalogObjectNameReference

### PR DESCRIPTION
**Pull request by aminelouati**
In this PR, we added:
1.  A new endpoint listCatalogObjectNameReference that lists catalog objects by kind
2. Some integration tests

### CODE REVIEW

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 1]
```diff
@@ -160,6 +162,22 @@ public CatalogObjectMetadataList create(
         return catalogObjectService.getContentTypes();
     }
 
+    @ApiOperation(value = "Lists catalog objects by kind")
+    @ApiResponses(value = { @ApiResponse(code = 401, message = "User not authenticated"),
+                            @ApiResponse(code = 403, message = "Permission denied") })
+    @RequestMapping(value = "/{kind}/references", method = GET, produces = "application/json")
+    @ResponseStatus(HttpStatus.OK)
+    public List<CatalogObjectNameReference> listCatalogObjectNameReference(
+            @ApiParam(value = "sessionID", required = false) @RequestHeader(value = "sessionID", required = false) String sessionId,
```
Comment (mboussaa) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: sessionID should not be required?

File: src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java [hunk 2]
```diff
@@ -160,6 +162,22 @@ public CatalogObjectMetadataList create(
...
+            @PathVariable String kind) {
+        return catalogObjectService.getCatalogObjectsNameReferenceByKind(kind)
+                                   .stream()
```
Comment (mboussaa) [inline src/main/java/org/ow2/proactive/catalog/rest/controller/CatalogObjectController.java]: Code to move to the service

File: src/main/java/org/ow2/proactive/catalog/service/RestApiAccessService.java [hunk 3]
```diff
@@ -89,6 +77,28 @@ public void checkAccessBySessionIdForBucketAndThrowIfDeclined(boolean sessionIdR
 
     }
 
+    public boolean isAPublicBucket(String bucketName) {
```
[the remark this answers is not in the document]
Comment (aminelouati) [reply src/main/java/org/ow2/proactive/catalog/service/RestApiAccessService.java]: Than you for the remark. My purpose is to group on top all public methods and do the same for the private methods.

**Comment by aminelouati** [comment]
re-test this please

**Comment by aminelouati** [comment]
re-test this please

**Comment by aminelouati** [comment]
re-test this please

**Comment by aminelouati** [comment]
re-test this please

**Comment by aminelouati** [comment]
re-test this please

**Comment by aminelouati** [comment]
re-test this please

**Pull request merged**